src/ga_tx.cpp

```diff
@@ -809,9 +805,41 @@ namespace sdk {
                         continue;
                     }
 
-                    // We have more than the dust amount of change. Add a change
-                    // output to collect it, then loop again in case the amount
-                    // this increases the fee by requires more UTXOs.
+                    // We have more than the dust amount of change. First look for an explicit change
+                    // output in the addressees and if present send the change there
+                    amount::value_type change_amount = (total - required_total - fee).value();
+                    bool have_explicit_change = false;
+                    if (num_addressees) {
+                        int addressee_index = 0;
+                        for (auto& addressee : *addressees_p) {
+                            const auto addressee_asset_tag
+                                = session.asset_id_from_string(addressee.value("asset_tag", std::string{}));
+                            if (addressee_asset_tag == asset_tag) {
+                                if (json_get_value(addressee, "is_change", false)) {
+                                    if (have_explicit_change) {
+                                        set_tx_error(result, "Only one explicit change addressee allowed");
+                                        break;
+                                    }
+                                    // There is an assumption that the tx outputs are indexed so they match
+                                    // addressees, which is very weakly enforced but needs to be true otherwise
+                                    // other things (liquid blinding) breaks.
+                                    set_tx_output_value(net_params, tx, addressee_index, asset_tag, change_amount);
+                                    addressee["satoshi"] = change_amount;
+                                    have_explicit_change = true;
+                                    required_total += change_amount;
+                                }
+                            }
+                            ++addressee_index;
+                        }
+                    }
+
+                    if (have_explicit_change) {
+                        // No need for implicit change output
+                        continue;
+                    }
+
```

Comment:
At 792 we should add the dust to an explicit change output if there is one.

I think we also need a check that if any outputs are marked `is_change` then the tx has no added change output already.

Reply:
I've moved this logic around a bit so that now any dust will be added to the greedy addressee before being considered for change.

Logic should be such that there won't be a change output if there is a greedy one.